servers/Azure.Mcp.Server/tests/Azure.Mcp.Server.UnitTests/Infrastructure/ConsolidatedModeTests.cs

```diff
@@ -0,0 +1,114 @@
+    {
+        // Arrange
+        var exeName = OperatingSystem.IsWindows() ? "azmcp.exe" : "azmcp";
+        var azmcpPath = Path.Combine(AppContext.BaseDirectory, "azmcp");
+
+        Assert.True(File.Exists(azmcpPath), $"Executable not found at {azmcpPath}. Please build the Azure.Mcp.Server project first.");
+
+        // Act - Start the server process with consolidated mode
+        var processStartInfo = new System.Diagnostics.ProcessStartInfo
+        {
+            FileName = azmcpPath,
+            Arguments = "server start --mode consolidated",
+            UseShellExecute = false,
+            RedirectStandardInput = true,
+            RedirectStandardOutput = true,
+            RedirectStandardError = true,
+            CreateNoWindow = true
+        };
+
+        using var process = System.Diagnostics.Process.Start(processStartInfo);
+        Assert.NotNull(process);
+
+        try
+        {
+            // Give the process a moment to start up
+            await Task.Delay(500, TestContext.Current.CancellationToken);
+
+            // Send initialize request first (required by MCP protocol)
+            var initRequest = """
+                {"jsonrpc":"2.0","id":1,"method":"initialize","params":{"protocolVersion":"2024-11-05","capabilities":{},"clientInfo":{"name":"test-client","version":"1.0"}}}
+                """;
+            await process.StandardInput.WriteLineAsync(initRequest);
+            await process.StandardInput.FlushAsync(TestContext.Current.CancellationToken);
+
+            // Read initialize response
+            var initResponse = await ReadJsonRpcResponseAsync(process.StandardOutput);
+            Assert.NotNull(initResponse);
+            Assert.Contains("\"result\"", initResponse, StringComparison.OrdinalIgnoreCase);
+
+            // Send initialized notification
+            var initializedNotification = """
+                {"jsonrpc":"2.0","method":"notifications/initialized"}
+                """;
+            await process.StandardInput.WriteLineAsync(initializedNotification);
+            await process.StandardInput.FlushAsync(TestContext.Current.CancellationToken);
+
+            // Send tools/list request
+            var listToolsRequest = """
+                {"jsonrpc":"2.0","id":2,"method":"tools/list","params":{}}
+                """;
+            await process.StandardInput.WriteLineAsync(listToolsRequest);
+            await process.StandardInput.FlushAsync(TestContext.Current.CancellationToken);
+
+            // Read tools/list response
+            var listToolsResponse = await ReadJsonRpcResponseAsync(process.StandardOutput);
+            Assert.NotNull(listToolsResponse);
+
+            // Assert - Verify we got tools back
+            Assert.Contains("\"result\"", listToolsResponse, StringComparison.OrdinalIgnoreCase);
+            Assert.Contains("\"tools\"", listToolsResponse, StringComparison.OrdinalIgnoreCase);
```

Comment:
Don't we need to do more validation about the tools we got back? Where is the actual validation?

Reply:
It is sufficient as long as the consolidated mode starts and can list tools.